pkg/venafi/cloud/connector.go

```diff
@@ -827,6 +987,96 @@ func (c *Connector) RetrieveSystemVersion() (response string, err error) {
 	panic("operation is not supported yet")
 }
 
+func setProvisioningOptions(options *endpoint.ProvisioningOptions) (*cloudkeystores.CertificateProvisioningOptionsInput, error) {
+	var cloudOptions *cloudkeystores.CertificateProvisioningOptionsInput
+	dataOptions, err := json.Marshal(options)
+	if err != nil {
+		return nil, err
+	}
+
+	graphqlAzureOptions := &cloudkeystores.CertificateProvisioningAzureOptionsInput{}
+	graphqlGCPOptions := &cloudkeystores.CertificateProvisioningGCPOptionsInput{}
+
+	if options != nil {
+		switch (*options).GetType() {
+		case string(cloudkeystores.CloudKeystoreTypeAcm):
+			// nothing
+		case string(cloudkeystores.CloudKeystoreTypeAkv):
+			err = json.Unmarshal(dataOptions, graphqlAzureOptions)
+			if err != nil {
+				return nil, err
+			}
+		case string(cloudkeystores.CloudKeystoreTypeGcm):
+			err = json.Unmarshal(dataOptions, graphqlGCPOptions)
+			if err != nil {
+				return nil, err
+			}
+		default:
+			return nil, fmt.Errorf("unknown cloud keystore type: %s", (*options).GetType())
+		}
+	}
+
+	cloudOptions = &cloudkeystores.CertificateProvisioningOptionsInput{
+		AwsOptions:   nil,
```

Comment:
If AWS does not have any options, you should remove the attribute from the struct all together

Reply:
This is from generated code:
type CertificateProvisioningOptionsInput struct {
    AwsOptions   *CertificateProvisioningAWSOptionsInput   `json:"awsOptions,omitempty"`
    AzureOptions *CertificateProvisioningAzureOptionsInput `json:"azureOptions,omitempty"`
    GcpOptions   *CertificateProvisioningGCPOptionsInput   `json:"gcpOptions,omitempty"`
}

from schema:
input CertificateProvisioningOptionsInput
@join__type(graph: CLOUD_PROVIDERS)
{
    awsOptions: CertificateProvisioningAWSOptionsInput
    azureOptions: CertificateProvisioningAzureOptionsInput
    gcpOptions: CertificateProvisioningGCPOptionsInput
}

There's no much we can do to remove it. As we already discussed offline. There is no futher action to do here so I'll just resolve this comment.